sql/updates/latest-dev.sql

```diff
@@ -118,3 +118,82 @@ ALTER TABLE IF EXISTS _timescaledb_catalog.compression_chunk_size ADD COLUMN IF
 CLUSTER  _timescaledb_catalog.compression_chunk_size USING compression_chunk_size_pkey;
 ALTER TABLE _timescaledb_catalog.compression_chunk_size SET WITHOUT CLUSTER;
 
+-- If the dimension slice table is broken and there are dimension
+-- slices missing from the table, we will repair it by:
+--    1. Finding all chunk constraints that have missing dimension
+--       slices and extract the constraint expression from the associated
+--       constraint.
+--    2. Parse the constraint expression and extract the column name,
+--       and upper and lower range values as text.
+--    3. Use the column type to construct the range values (UNIX
+--       microseconds) from these values.
+
+-- View with all dimension slices that are mentioned in the
+-- chunk_constraint table but are missing from the dimension_slice
+-- table.
+CREATE VIEW missing_slices(dimension_slice_id, constraint_name, constraint_expr) AS
+SELECT DISTINCT dimension_slice_id
+   	      , constraint_name
+	      , pg_get_expr(conbin,conrelid) AS expr
+       FROM _timescaledb_catalog.chunk_constraint
+  LEFT JOIN _timescaledb_catalog.dimension_slice sl
+         ON dimension_slice_id = sl.id
+       JOIN pg_constraint ON conname = constraint_name
+     WHERE dimension_slice_id IS NOT NULL AND sl.id IS NULL;
+
+-- View with the column name and unparsed lower and upper range for
+-- each dimension slice id that is missing.
+CREATE VIEW unparsed_missing_slices(dimension_slice_id, constraint_name, column_name, lower_range, upper_range) AS
+SELECT dimension_slice_id
+     , constraint_name
+     , COALESCE(SUBSTRING(constraint_expr, '(\w+)\s*(?:>=|<)'),
+                SUBSTRING(constraint_expr, '"([^"]+)"\s*(?:>=|<)'),
+		SUBSTRING(constraint_expr, 'get_partition_hash\((\w+)\)'),
+		SUBSTRING(constraint_expr, 'get_partition_hash\("([^"]+)"\)')) AS column_name
+     , (SELECT SUBSTRING(constraint_expr, $$>=\s*'?([\d\s:+-]+)'?$$)) AS lower_range
+     , (SELECT SUBSTRING(constraint_expr, $$<\s*'?([\d\s:+-]+)'?$$)) AS upper_range
+  FROM missing_slices;
+
+-- View with dimension slice id, dimension id, and start and end
+-- values for range as bigint for each dimension slice that is missing
+-- from the dimension_slice table. These rows are identical to what is
+-- stored in the dimension_slice table.
+CREATE VIEW rebuilt_missing_slices(dimension_slice_id, dimension_id, range_start, range_end) AS
+SELECT dimension_slice_id
+     , di.id AS dimension_id
+     , CASE
+       WHEN di.column_type IN ('smallint'::regtype, 'bigint'::regtype, 'integer'::regtype) THEN
```

Comment:
There's a function `_timescaledb_internal.time_to_internal(time_val ANYELEMENT)` that would probably help with this conversion as long as you first cast the value to the right type.